hack/prepopulate-kind-registry.sh

```diff
@@ -22,10 +23,34 @@ all_images=(
 )
 
 # Pull, tag, and push all images to local registry.
-for image in ${all_images[@]}; do
+# Docker will pull images from remote only if one is not available locally or is not upto date.
+# NOTE: We are pushing these images to local registry instead of loading to kind because
+#       these images either have tag "latest" or have the image pull policy "Always". In both of
+#       these cases, kind will pull new image even if one is available. So, we want kind to first
+#       try pulling these images from local registry, and if one is not available, pull from
+#       remote repository. 
+for image in ${images_to_push[@]}; do
     echo ""
     echo "> 🔨  Pull, tag, and push ${image}"
     docker pull ${image}
     docker tag ${image} localhost:5000/${image}
     docker push localhost:5000/${image}
 done
+
+images_to_load=(
+  "docker.io/istio/examples-bookinfo-details-v1:1.16.2"
+  "docker.io/istio/examples-bookinfo-productpage-v1:1.16.2"
+  "docker.io/istio/examples-bookinfo-ratings-v1:1.16.2"
+  "docker.io/istio/examples-bookinfo-reviews-v1:1.16.2"
+  "docker.io/datawire/aes:1.12.1"
+  "prom/statsd-exporter:v0.8.1"
+  "redis:5.0.1"
+)
+
+# Pull image if not available locally or is not upto date and then load to kind.
+for image in ${images_to_load[@]}; do
+    echo ""
+    echo "> 🔨  Pull, load to kind ${image}"
+    docker pull ${image}
+    kind load docker-image ${image} --name ${KIND_CLUSTER_NAME}
+done
```

Comment:
These images may be available when you run it on your local machine but the CI node (github actions VM) is not going to have them cached (unless there is a way to do this ??). In which case it would take the same time to pull images down to the local registry as it would when the first test is run, albeit slightly faster.

Reply:
I did this to speed up testing on local machine. Also, GitHub CI workflow is not running `make setup-dev-env` or `make dev`. So, this script will not run on GitHub CI node. Do you want me to pre-load images to kind node on GitHub CI. When on GitHub CI, we will only load images to kind node, and will skip local registry step.

Reply:
Let's build a KinD node image (https://kind.sigs.k8s.io/docs/design/node-image/) with the docker images preloaded in the image's docker registry (filesystem) and use that when we start the kind cluster instead of the base image in the Github CI. Does that make sense?

Reply:
Working on it; will let you know if I have any question.